container_registry/container_analysis/src/sample/samples_test.go

```diff
@@ -313,3 +258,166 @@ func TestPubSub(t *testing.T) {
 	sub.Delete(v.ctx)
 	teardown(t, v)
 }
+
+func TestPollDiscoveryOccurrenceFinished(t *testing.T) {
+	v := setup(t)
+
+	timeout := time.Duration(1) * time.Second
+	discOcc, err := pollDiscoveryOccurrenceFinished(v.imageUrl, v.projectID, timeout)
+	if err == nil || discOcc != nil {
+		t.Errorf("expected error when resourceUrl has no discovery occurrence")
+	}
+
+	// create discovery occurrence
+	noteId := "discovery-note-" + v.timestamp
```

Comment:
We've run into problems in the past with relying on timestamps to be unique. We run the tests in parallel and it's surprisingly common how often two different runs use the same timestamp.

It's safer if they only need to be unique per-project. But, consider using the `uuid` package to come up with a unique value if needed. Even better if it doesn't rely on uniqueness at all (not sure if that's possible). :)

Reply:
This one does only need to be unique per-project, but I replaced it with a uuid anyway